#167164772 Feature Common Components

despeauxz opened the pull request:
#### What does this PR do?
This PR adds some common components relative to the entire application
#### Description of Task to be completed?
Create the following components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader
#### How should this be manually tested?
Navigate through the `<rootDir>/src/components/commons`
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#167164772
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/16857803/61012905-5640c880-a378-11e9-9039-eadf77c8235b.png)
![image](https://user-images.githubusercontent.com/16857803/61012910-5e990380-a378-11e9-84d1-1a89e1e9d84b.png)
![image](https://user-images.githubusercontent.com/16857803/61012921-6658a800-a378-11e9-8e01-686ed6c2e0fc.png)
<img width="1440" alt="Screenshot 2019-07-10 at 6 12 56 PM" src="https://user-images.githubusercontent.com/20377428/61040147-f83bd180-a3c7-11e9-98a3-512bf133c8a5.png">
<img width="1440" alt="Screenshot 2019-07-10 at 3 04 28 PM" src="https://user-images.githubusercontent.com/20377428/61040148-f8d46800-a3c7-11e9-8f56-348b48dae403.png">
<img width="1440" alt="Screenshot 2019-07-10 at 3 05 18 PM" src="https://user-images.githubusercontent.com/20377428/61040150-f8d46800-a3c7-11e9-8d21-e388fad8f02e.png">
<img width="1440" alt="Screenshot 2019-07-10 at 6 13 50 PM" src="https://user-images.githubusercontent.com/20377428/61040151-f96cfe80-a3c7-11e9-9889-518b1d504c86.png">
<img width="1440" alt="Screenshot 2019-07-10 at 6 12 45 PM" src="https://user-images.githubusercontent.com/20377428/61040153-f96cfe80-a3c7-11e9-9e0b-f4d46cca4270.png">


#### Questions:
N/A

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configured the store
- initialized an empty reducer

[#166272289]

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
chore-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
Merge branch 'develop' into 166272289-ch-setup-redux

despeauxz committed:
chore-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

Finishes #167164772]

despeauxz committed:
chore-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader
[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

fidelisojeah commented on src/components/commons/FormComponents/RenderInput.js:
```diff
@@ -0,0 +1,68 @@
+            name={name}
+            className={`${className} w-full py-2 border-gray-600 border-b-2 outline-none`}
+            value={value}
+            onChange={e => handleChange(e)}
```
Don't do this.

do this instead
```suggestion
            onChange={handleChange}
```

fidelisojeah commented on src/components/commons/Header/index.jsx:
```diff
@@ -0,0 +1,200 @@
+          <button
+            type='button'
+            className='text-blue-600 cursor-pointer'
+            onClick={this.showSearchBar.bind(this)}
```
Don't bind here

fidelisojeah commented on src/components/commons/Header/index.jsx:
```diff
@@ -0,0 +1,200 @@
+    </Fragment>
+  );
+
+  showSearchBar() {
```
```suggestion
  showSearchBar = () => {
```

This binds for you.

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]
